src/components/NameValueInput.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FC } from 'react';
+import { css } from 'emotion';
+import { HorizontalGroup, Input, IconButton, VerticalGroup, Icon, Button, Field, useTheme } from '@grafana/ui';
+import { useFieldArray, useFormContext } from 'react-hook-form';
+
+interface Props {
+  name: string;
+  limit: number;
+  disabled?: boolean;
+  label: string;
+  validateName?: (name: string) => string | undefined;
+  validateValue?: (value: string) => string | undefined;
+}
+
+export const NameValueInput: FC<Props> = ({ name, disabled, limit, label, validateName, validateValue }) => {
+  const { register, control, errors } = useFormContext();
+  const { fields, append, remove } = useFieldArray({ control, name });
+  const theme = useTheme();
+  const fieldError = name.split('.').reduce((error, current) => error?.[current], errors);
```

Comment:
clever!

Reply:
Yeah but kind of a brain twister due to relying on implicit behavior and using `reduce` which doesn't really resemble what's actually going on. Which... I'm btw not completely sure about.

But it looks like you're probably not actually using all parts of `name` but instead rely on implicit discarding of irrelevant parts of it. And then use `reduce` to extract the error that you want if there is one.

So would this work as well?
```
const fieldError = errors?.[name.split('.').slice(-1)[0]];
```

Reply:
Was just wondering whether the readability could be helped along by for instance renaming `error` to `nestedError` and `current` to `errorPathFragment`.